### PR TITLE
Bug 1928932: Update deploy manifests to use stable apis

### PR DESCRIPTION
**Pull request by matthyx**
https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/

### CODE REVIEW

**Comment by matthyx** [comment]
Can someone have a look at this?

**Comment by christianh814** [comment]
I ran into an issue today withe the deployment YAMLs as well.

File: deploy/route_crd.yaml [hunk 1]
```diff
@@ -13,28 +13,29 @@ spec:
...
+          type: string
+          jsonPath: .spec.to.name
+        - name: TLS
+          type: string
+          jsonPath: .spec.tls.type
   # either Namespaced or Cluster
   scope: Namespaced
-  subresources:
-    # enable spec/status
-    status: {}
```
Comment (Miciah) [inline deploy/route_crd.yaml:20]: Why is `subresources:` deleted?  The CRD should still define a `status` subresource.

Comment (matthyx) [reply deploy/route_crd.yaml]: Correct, I have removed it cluelessly. It's back

**Comment by christianh814** [comment]
Just tested your branch @matthyx and I'm still getting an error

```shell
error: error validating "router/deploy/route_crd.yaml": error validating data: ValidationError(CustomResourceDefinition.spec): unknown field "subresources" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.CustomResourceDefinitionSpec; if you choose to ignore these errors, turn validation off with --validate=false
```
This is what ultimately worked for me

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  # name must match the spec fields below, and be in the form: <plural>.<group>
  name: routes.route.openshift.io
spec:
  # group name to use for REST API: /apis/<group>/<version>
  group: route.openshift.io
  # list of versions supported by this CustomResourceDefinition
  versions:
    - name: v1
      # Each version can be enabled/disabled by Served flag.
      served: true
      # One and only one version must be marked as the storage version.
      storage: true
      schema:
        openAPIV3Schema:
          type: object
          x-kubernetes-preserve-unknown-fields: true
      additionalPrinterColumns:
        - name: Host
          type: string
          jsonPath: .status.ingress[0].host
        - name: Admitted
          type: string
          jsonPath: .status.ingress[0].conditions[?(@.type=="Admitted")].status
        - name: Service
          type: string
          jsonPath: .spec.to.name
        - name: TLS
          type: string
          jsonPath: .spec.tls.type
      subresources:
        # enable spec/status
        status: {}
  # either Namespaced or Cluster
  scope: Namespaced
  names:
    # plural name to be used in the URL: /apis/<group>/<version>/<plural>
    plural: routes
    # singular name to be used as an alias on the CLI and for display
    singular: route
    # kind is normally the CamelCased singular type. Your resource manifests use this.
    kind: Route
```

**Comment by Miciah** [comment]
/retitle Bug 1928932: Update deploy manifests to use stable apis
/approve

**Comment by matthyx** [comment]
> Just tested your branch @matthyx and I'm still getting an error

Ah yes, indentation... I just edited my branch in codium and didn't connect to the office cluster to check.
Thanks for the fix, I think scope is namespaced by default?

**Review by christianh814 — APPROVED** [review]
/lgtm

**Comment by matthyx** [comment]
@christianh814 can you `/retest` please?

**Pull request merged**